### PR TITLE
Fallback to 3.0 for later swift version as a backup

### PR DESCRIPTION
**Pull request by wuhao5**

### CODE REVIEW

**Comment by wuhao5** [comment]
@mettieDie @bootstraponline can you take a look? thanks!

**Comment by tirodkar** [comment]
@wuhao5 win't you need https://github.com/google/EarlGrey/pull/581 along with this? We should keep that open just for the Swift file updates.

File: gem/lib/earlgrey/configure_earlgrey.rb [hunk 1]
```diff
@@ -378,7 +378,10 @@ def copy_swift_files(project, target, swift_version = nil)
       src_swift_name = 'EarlGrey.swift'
       src_swift = File.join(src_root, "Swift-#{swift_version}", src_swift_name)
 
-      raise 'Bundled swift missing' unless File.exist? src_swift
+      unless File.exist? src_swift
+        puts_magenta "Fallback to swift version 3.0 for #{swift_version}."
```
Comment (bootstraponline) [inline gem/lib/earlgrey/configure_earlgrey.rb]: I think this should say something like:

> "Swift version #{swift_version} not found. Falling back to 3.0"

File: gem/lib/earlgrey/configure_earlgrey.rb [hunk 2]
```diff
@@ -378,7 +378,10 @@ def copy_swift_files(project, target, swift_version = nil)
...
+        src_swift = File.join(src_root, "Swift-3.0", src_swift_name)
```
Comment (bootstraponline) [inline gem/lib/earlgrey/configure_earlgrey.rb]: should probably raise here:

```ruby
swift_fallback = 'Swift-3.0'
raise "Unable to locate #{swift_fallback} file" unless File.exist?(src_swift)
```

**Comment by wuhao5** [comment]
> win't you need #581 along with this?

No, let's try to not have many swift files to maintain. hopefully once swift abi is stable and compatible, we can bake it into the framework.

**Comment by wuhao5** [comment]
@bootstraponline please take one more look!

File: gem/lib/earlgrey/configure_earlgrey.rb [hunk 3]
```diff
@@ -378,7 +378,12 @@ def copy_swift_files(project, target, swift_version = nil)
       src_swift_name = 'EarlGrey.swift'
       src_swift = File.join(src_root, "Swift-#{swift_version}", src_swift_name)
 
-      raise 'Bundled swift missing' unless File.exist? src_swift
+      unless File.exist? src_swift
+        puts_magenta "Swift version #{swift_version} not found, falling back to 3.0."
```
Comment (khandpur) [inline gem/lib/earlgrey/configure_earlgrey.rb]: EarlGrey.swift for version #{swift_version} not found. Falling back to version 3.0.

File: gem/lib/earlgrey/configure_earlgrey.rb [hunk 4]
```diff
@@ -378,7 +378,12 @@ def copy_swift_files(project, target, swift_version = nil)
...
+        swift_fallback = 'Swift-3.0'
+        src_swift = File.join(src_root, swift_fallback, src_swift_name)
+        raise "Unable to locate #{swift_fallback} file" unless File.exist?(src_swift)
```
Comment (tirodkar) [inline gem/lib/earlgrey/configure_earlgrey.rb]: Unable to locate #{swift_fallback} file at path @{src_swift}.

**Comment by bootstraponline** [comment]
> hopefully once swift abi is stable

```
Development major version: Swift 5.0
Expected release date: Late 2018
Primary Focus: ABI Stability

https://github.com/apple/swift-evolution
```

looks like 2018/2019 at the earliest. 😭 

The changes look good!

**Comment by wuhao5** [comment]
PTAL!

**Comment by khandpur** [comment]
Fixes #580

**Pull request merged**